src/components/Datepicker/Datepicker.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import { Button, ButtonIcon } from 'react-lds';
+import classnames from 'classnames';
+
+import moment from 'moment';
+import 'moment-range';
+
+import { prefixClasses } from '../../utils';
+
+export class Datepicker extends React.Component {
+  static contextTypes = { cssPrefix: React.PropTypes.string };
+
+  static propTypes = {
+    /**
+    * Optional preselected date
+    */
+    selectedDate: React.PropTypes.string,
+    /**
+    * Callback function once day has been changed
+    */
+    onSelectDate: React.PropTypes.func.isRequired,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.prefix = (classes, passThrough) => prefixClasses(this.context.cssPrefix, classes, passThrough);
+
+    this.state = {
+      selectedDate: this.props.selectedDate ? moment(this.props.selectedDate) : null,
+      viewedDate: moment(),
+    };
+
+    this.viewYear = (e) => {
+      this.setState({ viewedDate: this.state.viewedDate.set('year', e.target.value) });
+    };
+    this.viewMonth = (relativeMonth) => {
+      this.setState({ viewedDate: this.state.viewedDate.add(relativeMonth, 'month') });
+    };
+    this.selectDay = (day) => {
+      this.setState({ selectedDate: day });
+      this.props.onSelectDate(day.format('YYYY-MM-DD'));
+    };
+    this.resetViewToToday = () => this.setState({ viewedDate: moment() });
+  }
+
+  getMonthDays() {
+    // Construct range consisting of all days in month and another range consisting
+    // of all days of all weeks which the month is in
+    const firstDay = moment(this.state.viewedDate).startOf('month');
+    const lastDay = moment(this.state.viewedDate).endOf('month');
+    const firstDayFirstWeek = moment(firstDay).startOf('week');
+    const lastDayLastWeek = moment(lastDay).endOf('week');
+
+    // Create two ranges, one spanning all weeks touched by the current month,
+    // and one spanning only the month
+    const monthRange = moment.range(firstDay, lastDay);
+    const fullWeekRange = moment.range(firstDayFirstWeek, lastDayLastWeek);
+
+    // Return filled week bins
+    return fullWeekRange.toArray('days').reduce((acc, val) => {
+      if (moment(val).day() % 7 === 0) { acc.push([]); }
+      acc[acc.length - 1].push([moment(val), moment(val).within(monthRange)]);
+      return acc;
+    }, [[]]).filter(bin => bin.length > 0);
+  }
+
+  renderYearOptions() {
+    const prevYears = moment(this.state.viewedDate).subtract(2, 'years');
```

Comment:
the amount of years before and after the current year should be made configurable via a prop